app/controllers/allocations_controller.rb

```diff
@@ -12,8 +12,18 @@ def upload
     file = params[:csv].tempfile
     csv = CSV.read(file)
     group = Group.find(params[:group_id])
-    AllocationService.create_allocations_from_csv(csv: csv, group: group, current_user: current_user)
-    render nothing: true, status: 200
+    errors = []
+    ActiveRecord::Base.transaction do
+      errors = AllocationService.create_allocations_from_csv(csv: csv, group: group, current_user: current_user)
+      if errors
+        raise ActiveRecord::Rollback
+      end
+    end
```

Comment:
could we move the `ActiveRecord` stuff into `AllocationService` ?

i'd like to keep our controllers as lean as possible, somethin like

```ruby
file = params[:csv].tempfile
csv = CSV.read(file)
group = Group.find(params[:group_id])

if errors = AllocationService.create_allocation_from_csv(csv: csv, group: group, current_user: current_user)
  render json: { errors: errors }, status: 409
else
  render nothing: true, status: 200
end
```